crates/aptos/src/common/transactions.rs

```diff
@@ -0,0 +1,343 @@
+
+        // Warn local user that clock is skewed behind the blockchain.
+        // There will always be a little lag from real time to blockchain time
+        if now_usecs < state.timestamp_usecs - ACCEPTED_CLOCK_SKEW_US {
+            eprintln!("Local clock is is skewed from blockchain clock.  Clock is more than {} seconds behind the blockchain {}", ACCEPTED_CLOCK_SKEW_US, state.timestamp_usecs / US_IN_SECS );
+        }
+        let expiration_time_secs = now + self.gas_options.expiration_secs;
+
+        let chain_id = ChainId::new(state.chain_id);
+
+        let transaction_factory =
+            TransactionFactory::new(chain_id).with_gas_unit_price(gas_unit_price);
+
+        let unsigned_transaction = transaction_factory
+            .payload(payload.clone())
+            .sender(sender_address)
+            .sequence_number(sequence_number)
+            .expiration_timestamp_secs(expiration_time_secs)
+            .build();
+
+        // TODO: Support other transaction authenticator types, like multi-agent and fee-payer
+        let signed_transaction = SignedTransaction::new_signed_transaction(
+            unsigned_transaction,
+            TransactionAuthenticator::SingleSender {
+                sender: AccountAuthenticator::NoAccountAuthenticator,
+            },
+        );
+
+        let simulated_txn = client
+            .simulate_bcs_with_gas_estimation(&signed_transaction, true, false)
+            .await?
+            .into_inner();
+
+        let user_txn = simulated_txn.transaction.try_as_signed_user_txn().unwrap();
+
+        // TODO: add events and outputs
+        Ok(TransactionSummary {
+            transaction_hash: simulated_txn.info.hash().into(),
+            gas_used: Some(simulated_txn.info.gas_used()),
+            gas_unit_price: Some(user_txn.gas_unit_price()),
+            pending: None,
+            sender: Some(user_txn.sender()),
+            sequence_number: Some(user_txn.sequence_number()),
+            success: Some(simulated_txn.info.status().is_success()),
+            timestamp_us: None,
+            version: Some(simulated_txn.version),
+            vm_status: Some(format!("{:?}", simulated_txn.info.status())), // TODO: add proper status
+        })
+    }
+
+    /// Simulates a transaction locally, using the debugger to fetch required data from remote.
+    async fn simulate_using_debugger<F>(
+        &self,
+        payload: TransactionPayload,
+        execute: F,
+    ) -> CliTypedResult<TransactionSummary>
+    where
+        F: FnOnce(
+            &AptosDebugger,
+            u64,
+            SignedTransaction,
+            aptos_crypto::HashValue,
+        ) -> CliTypedResult<(VMStatus, VMOutput)>,
+    {
+        let client = self.rest_client()?;
+
+        // Fetch the chain states required for the simulation
+        // TODO(Gas): get the following from the chain
+        const DEFAULT_GAS_UNIT_PRICE: u64 = 100;
+        const DEFAULT_MAX_GAS: u64 = 2_000_000;
```

Comment:
A variant of the args where these args are non-optional would maybe be better, then they can see what values are being used from the CLI help message. But nbd, I'd understand if you wanna keep it this way to cut down on the different args variants.

Reply:
Oh, that's super old stuff right.  What happens when you learn how to do things the right way.

Maybe I'll clean up some of these, but yeah scope...